src/docsfy/renderer.py

```
@@ -58,16 +60,32 @@ def _sanitize_html(html: str) -> str:
+        if decoded.startswith("/") and not decoded.startswith("//"):
+            return True
+        # Reject protocol-relative URLs (//evil.com)
+        if decoded.startswith("//"):
+            return False
+        # Allow relative URLs (no scheme) - e.g., "page-slug.html", "page.html#section:details"
+        # Use urlsplit to check for a scheme — it correctly handles
+        # colons in fragments/paths vs actual schemes
+        parsed = urllib.parse.urlsplit(decoded)
+        if not parsed.scheme:
+            return True  # No scheme = relative URL, safe
+        return False
```

Ruff (0.15.10)

[warning] Return the condition directly

(SIM103)